### PR TITLE
#30: Remove the use of dynamic memory

### PR DESCRIPTION
**Pull request by TheSomeMan**

### CODE REVIEW

File: src/http_server.c [hunk 1]
```diff
@@ -88,40 +90,27 @@ static const char TAG[] = "http_server";
...
-    ESP_LOGD(TAG, "Respond: %s", buf);
+    ESP_LOGD(TAG, "Respond: %s", str_buf.buf);
     uint8_t netconn_flags = (uint8_t)NETCONN_COPY;
     if (flag_more)
     {
         netconn_flags |= (uint8_t)NETCONN_MORE;
     }
-    netconn_write(conn, buf, len, netconn_flags);
-    free(buf);
+    netconn_write(conn, str_buf.buf, len, netconn_flags);
```
Comment (ojousima) [inline src/http_server.c:112]: str_buf.size?

**Comment by ojousima** [comment]
Does this branch really remove dynamic memory use or just wrap mallocs in esp_wrapper functions? Of course those esp_wrappers could later be modified to use statically allocated blocks, but I'm not sure if that's really necessary.

**Comment by TheSomeMan** [comment]
> Does this branch really remove dynamic memory use or just wrap mallocs in esp_wrapper functions? Of course those esp_wrappers could later be modified to use statically allocated blocks, but I'm not sure if that's really necessary.

it does only in first commit (http_server). In all subsequent commits mallocs was replaced by static memory.

**Review by ojousima — APPROVED** [review]
Thanks

**Pull request merged**